src/Microsoft.DotNet.MSBuildSdkResolver/EnvironmentProvider.cs

```diff
@@ -21,21 +20,25 @@ public EnvironmentProvider(Func<string, string> getEnvironmentVariable)
             _getEnvironmentVariable = getEnvironmentVariable;
         }
 
-        public IEnumerable<string> ExecutableExtensions
+        public string ExecutableExtension
         {
             get
             {
-                if (_executableExtensions == null)
-                {
-
-                    _executableExtensions = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
-                        ? _getEnvironmentVariable("PATHEXT")
-                            .Split(';')
-                            .Select(e => e.ToLower().Trim('"'))
-                        : new [] { string.Empty };
-                }
+                return IsWindows ? ".exe" : string.Empty;
+            }
+        }
 
-                return _executableExtensions;
+        private bool IsWindows
+        {
+            get
+            {
+#if NET46
+                var platform = (int)Environment.OSVersion.Platform;
+                var isWindows = (platform != 4) && (platform != 6) && (platform != 128);
```

Comment:
These magic numbers are very troubling.

The NET46 build already assumes windows so we can just return ".exe". I'd refactor IsWindows into the Interop class that is already split by TFM and have NET46 return true and NETSTANDARD return RuntimeInformation.IsOSPlatform(OSPlatform.Windows). Then, we should move the existing check in Interop.netstandard for Windows to call Interop.IsWindows for consistency.